File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int maxW = 0;
        int maxH = 0;

        for(int[] arr : sizes){
            int[] sortedArr = Arrays.stream(arr).sorted().toArray();
            maxH = Math.max(maxH, sortedArr[0]);
            maxW = Math.max(maxW, sortedArr[1]);
        }

        return maxH * maxW;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
import kotlin.math.max
import kotlin.math.min

class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var maxN = 0
        var minN = 0
        for (arr in sizes){
            val s = arr.sortedDescending()
            maxN = max(maxN, s[0])
            minN = max(minN, s[1])
        }

        return maxN * minN
    }
}